Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs

```diff
@@ -92,10 +92,18 @@ static class ModuleCreator {
 		static Func<DmdLazyMetadataBytes> CreateDynamicGetMetadataDelegate(DbgEngineImpl engine, DnModule dnModule) {
 			Debug.Assert(dnModule.IsDynamic);
 			var comMetadata = dnModule.CorModule.GetMetaDataInterface<IMetaDataImport2>();
-			if (comMetadata is null)
-				throw new InvalidOperationException();
-			var result = new DmdLazyMetadataBytesCom(comMetadata, engine.GetDynamicModuleHelper(dnModule), engine.DmdDispatcher);
-			return () => result;
+			if (comMetadata == null) {
+				if (comMetadata is null && dnModule.Process.CorProcess.CLRVersion.Major != 6) {
+					throw new InvalidOperationException();
+				}
+				else if (comMetadata is null && dnModule.Process.CorProcess.CLRVersion.Major == 6) //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
+					return () => new DmdLazyMetadataBytesNull();
```

Comment:
The null check is duplicated: `comMetadata == null` L95 and `comMetadata is null` in L95 and L99.
```suggestion
			if (comMetadata is null) {
				if (dnModule.Process.CorProcess.CLRVersion.Major != 6) {
					throw new InvalidOperationException();
				}
				else if (dnModule.Process.CorProcess.CLRVersion.Major == 6) //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
					return () => new DmdLazyMetadataBytesNull();
```